_rules/non-visual-reference-alternative-9bd38c.md

```diff
@@ -0,0 +1,441 @@
+
+This paragraph includes the [visual reference word][] "right". The user is told to find the navigation on the right and the navigation is also correctly identified by a `nav` element whose [accessible name][] contains the word "navigation", thus matching the (**non-visual reference**) condition.
+
+```html
+<head>
+	<title>Passed example 13 9bd38c</title>
+	<link rel="stylesheet" type="text/css" href="../test-assets/9bd38c/columns.css" />
+</head>
+<body>
+	<div class="col-container">
+		<div class="col">
+			<p>Find the navigation on the right, for the non-essential links</p>
+		</div>
+		<nav>
+			<div class="col">
+				<ul>
+					<li>
+						<a href="https://www.w3.org/Consortium/contact">Contact</a>
+					</li>
+					<li>
+						<a href="https://www.w3.org/Help/">Help and FAQ</a>
+					</li>
+				</ul>
+			</div>
+		</nav>
+	</div>
+</body>
+```
+
+#### Passed Example 14
+
+No [text node][] on this page includes any of the [visual reference words][].
+
+```html
+<p>Click the button, for a surprise</p>
+<button onclick="alert('Surprise!')">Howdy</button>
+```
+
+#### Passed Example 15
+
+The [text node][] inside the `p` element is not applicable because it is neither [visible][] nor [included in the accessibility tree][] due to its `display:none` styling. The [text node][] inside the `button` element does not include any of the [visual reference words][].
+
+```html
+<p style="display:none">Click the box, for a surprise</p>
+<button onclick="alert('Surprise!')">Howdy</button>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This paragraph includes the [visual reference word][] "right". The user is told to find the menu on the right but the menu is not identified in any other way.
+
+```html
+<head>
+	<title>Failed example 1 9bd38c</title>
+	<link rel="stylesheet" type="text/css" href="../test-assets/9bd38c/columns.css" />
+</head>
+<body>
+	<div class="col-container">
+		<div class="col">
+			<p>Find the menu on the right, to navigate</p>
+		</div>
+		<div class="col">
+			<ul>
+				<li>
+					<a href="https://www.w3.org/Consortium/contact">Contact</a>
+				</li>
+				<li>
+					<a href="https://www.w3.org/Help/">Help and FAQ</a>
+				</li>
+			</ul>
+		</div>
+	</div>
+</body>
+```
+
+#### Failed Example 2
+
+This paragraph includes the [visual reference word][] "right". The user is told to find the navigation on the right and the navigation is correctly identified by a `nav` element, but there are 2 `nav` elements on the page so the user doesn't know which one to use (the non-visual reference is ambiguous and does not identify a specific [web content][]).
+
+```html
+<head>
+	<title>Failed example 2 9bd38c</title>
+	<link rel="stylesheet" type="text/css" href="../test-assets/9bd38c/columns.css" />
+</head>
+<body>
+	<nav>
+		<ul>
+			<li>
+				<a href="https://www.w3.org/">W3C homepage</a>
+			</li>
+			<li>
+				<a href="https://www.w3.org/standards/">Standards</a>
+			</li>
+		</ul>
+	</nav>
+	<div class="col-container">
```

Comment:
According to the reading order user know that I am instructed to find navigation that is visually on the right of the test target that is actually the instructed navigation is both visually and programmatically (right-handed navigation is under col-container then problem col etc) on the right side of the test target. How this example is failed according to the applicability and expectation of this rule? Can you please a bit more?

Reply:
That's a bit of a question of whether "right" can be assumed to mean "further in reading order" or not.
This is a bit more controversial than "below" because there are many right-to-left scripts (I don't think there is any bottom-up script).
We can argue that English is left-to-right, so here "right" would mean "further in reading order", but we can also argue that non-native readers do not have that equivalence in mind and could be confused.

Moreover, in a independent columns layout like this one, I personally wouldn't understand that the column on the right is the sequel of the other. There is no real natural reading order between the text column and the menu column, so "right" might happen to be further in the DOM but I wouldn't undertand it as "further in reading order".